include/ginkgo/core/distributed/matrix.hpp

```diff
@@ -155,6 +155,22 @@ template <typename ValueType>
 class Vector;
 
 
+// TODO: move the data into this class
+template <typename IndexType>
+class MatrixBase {
```

Comment:
I think it is more appropriate to move the getters for the communication data (and subsequently the data itself) into its own class instead of the base class approach. I was working on something similar for the local matrix stuff.